docs/csharp/lambda-expressions.md

```diff
@@ -1,27 +1,160 @@
+
+The compiler can infer the type of the input parameter, or you can also specify it explicitly. This particular lambda expression counts those integers (`n`) that, when divided by two, have a remainder of 1.
+
+The following example produces a sequence that contains all elements in the `numbers` array that precede the 9, because that's the first number in the sequence that doesn't meet the condition.
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/query1.cs#5)]
+
+The following example specifies multiple input parameters by enclosing them in parentheses. The method returns all the elements in the numbers array until it encounters a number whose value is less than its ordinal position in the array.
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/query1.cs#6)]
+
+## Type inference in lambda expressions ##
+
+When writing lambdas, you often do not have to specify a type for the input parameters because the compiler can infer the type based on the lambda body, the parameter’s delegate type, and other factors, as described in the C# Language Specification. For most of the standard query operators, the first input is the type of the elements in the source sequence. So if you are querying an `IEnumerable<Customer>`, then the input variable is inferred to be a `Customer` object, which means you have access to its methods and properties:
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/infer1.cs#1)]
+
+The general rules for type inteference for lambdas are:
+
+- The lambda must contain the same number of parameters as the delegate type.
+
+- Each input argument in the lambda must be implicitly convertible to its corresponding delegate parameter.
+
+- The return value of the lambda (if any) must be implicitly convertible to the delegate's return type.
+
+Note that lambda expressions in themselves do not have a type because the common type system has no intrinsic concept of "lambda expression." However, it is sometimes convenient to speak informally of the "type" of a lambda expression. In these cases the type refers to the delegate type or @System.Linq.Expressions.Expression type to which the lambda expression is converted.
+
+## Variable Scope in Lambda Expressions ##
+
+Lambdas can refer to *outer variables* (see [Anonymous methods](programming-guide/statements-expressions-operators/anonymous-methods.md)) that are in scope in the method that defines the lambda function, or in scope in the type that contains the lambda expression. Variables that are captured in this manner are stored for use in the lambda expression even if the variables would otherwise go out of scope and be garbage collected. An outer variable must be definitely assigned before it can be consumed in a lambda expression. The following example demonstrates these rules.
```

Comment:
No, this should stay "definitely assigned". That's the phrase the spec always uses for the analysis of assignment to a variable before it is used.